src/protocols/output_management.rs

```diff
@@ -697,22 +705,22 @@ pub fn notify_changes(state: &mut OutputManagementManagerState, new_out: IpcOutp
                 }
                 (Some(_), None) => {
                     changed = true;
-                    for (_, client) in &state.clients {
-                        if let Some((head, _)) = client.heads.get(name) {
+                    for (_, client) in &g_state.clients {
+                        if let Some((head, _)) = client.heads.get(output) {
                             head.enabled(0);
                         }
                     }
                 }
                 (None, Some(new_index)) => {
                     changed = true;
-                    for (_, client) in &state.clients {
-                        if let Some((head, _)) = client.heads.get(name) {
+                    for (_, client) in &g_state.clients {
+                        if let Some((head, _)) = client.heads.get(output) {
                             head.enabled(1);
-                            if old.modes.len() <= new_index || new.modes.len() <= new_index {
+                            if old.modes.len() <= new_index || conf.modes.len() <= new_index {
                                 error!("notify_changes: out of bound mode");
-                            } else if old.modes[new_index] == new.modes[new_index] {
-                                for (_, client) in &state.clients {
-                                    if let Some((head, modes)) = client.heads.get(name) {
+                            } else if old.modes[new_index] == conf.modes[new_index] {
```

Comment:
Also should be a `old.modes == conf.modes` check with an error!()

Reply:
the whole mode check was a bit of a mess, mb.